### ISSUE TITLE
Should print warning message if type is undefined

### ISSUE THREAD

**Issue by LabiKyo** [acdlite/redux-actions#86]
Consider this code, it's commonly used in my project:
```javascript
import * as actionTypes from './actionTypes.js'
export default handleActions({
  [actionTypes.ACT_1]: (state, action) => {},
  [actionTypes.ACT_2]: (state, action) => {},
})
```

If there's a typo in actionTypes variable(`ACE_1` for instance), there's no warning, linters can't tell the error too. But it's easy to print warning message using redux-actions.

I'd like to send a PR if it's okay to you.

**Comment by timche** [comment]
If `console` messages would become part of `redux-actions`, then we also need an environment variable to not include them in production builds as they aren't necessary then. But I think this feature isn't needed, because this is personal preference how people are important the action types. I'll close this issue, but it can be reopened if necessary.

**Comment by LabiKyo** [comment]
Taking React as an example. React print warning messages

**Comment by LabiKyo** [comment]
Agree about environment variable. It's very common in libraries.

I do think it's a necessary feature. Taking React as an example. React print warning messages in dev env for propTypes checking and many other common cases. Because we are sure this is an error -- I don't know when we should handle an undefined actionType.

**Comment by timche** [comment]
So, are we using

**Comment by timche** [comment]
So, how about using `__DEV__` as env variable then? It's also what React and React Native are using.

**Comment by LabiKyo** [comment]
Haven't think about `createAction` and `handleAction`, and you're right about it.

For React, check [this](http://facebook.github.io/react/downloads.html#npm). I know when building react, I need to set `process.env.NODE_ENV` to `'production'`. Maybe React using `__DEV__` internally, need to check the code.

I'll finish PR in next week.

**Comment by Restuta** [comment]
Makes sense to me, NODE_ENV seems to be a standard thing to do, it doesn't matter what react uses internally, what matters is what is recommended to set outside and it's NODE_ENV.

**Comment by timche** [comment]
@yangmillstheory we could add this, because we've added `invariant`, what do you think?

**Comment by yangmillstheory** [comment]
Yes, let's do it! I can work on it.
